step-templates/apollo-rover-publish-subgraph-schema.json

```diff
@@ -0,0 +1,100 @@
+{
+  "Id": "51536440-5e87-4cc7-a254-acb673f9c2c8",
+  "Name": "Apollo Rover - Publish Subgraph Schema",
+  "Description": "Publish subgraph schema to Apollo Studio. This should be run after successfully deploying a subgraph service.\n\nThe script installs the [Apollo Rover CLI](https://apollographql.com/docs/rover) and runs the `rover subgraph publish` command.",
+  "ActionType": "Octopus.Script",
+  "Version": 1,
+  "CommunityActionTemplateId": null,
+  "Packages": [
+    {
+      "Name": "SchemaPackage",
+      "PackageId": null,
+      "FeedId": "feeds-builtin",
+      "AcquisitionLocation": "Server",
+      "Properties": {
+        "Extract": "True",
+        "SelectionMode": "deferred",
+        "PackageParameterName": "ApolloSchemaPackage",
+        "Purpose": ""
+      }
+    }
+  ],
+  "Properties": {
+    "Octopus.Action.Script.ScriptSource": "Inline",
+    "Octopus.Action.Script.Syntax": "Bash",
+    "Octopus.Action.Script.ScriptBody": "APOLLO_KEY=$(get_octopusvariable \"ApolloKey\")\nAPOLLO_GRAPH_REF=$(get_octopusvariable \"ApolloGraphRef\")\nSUBGRAPH_NAME=$(get_octopusvariable \"ApolloSubgraphName\")\nSUBGRAPH_URL=$(get_octopusvariable \"ApolloSubgraphUrl\")\nSCHEMA=$(get_octopusvariable \"ApolloSchema\")\nver=$(get_octopusvariable \"ApolloRoverVersion\")\n\n[ -z \"$ver\" ] || [ \"$ver\" == \"latest\" ] && ROVER_VERSION=\"latest\" || ROVER_VERSION=\"v$ver\"\n\nmissing_params=()\n\n[ -z \"$APOLLO_KEY\" ] && missing_params+=(\"ApolloKey\")\n[ -z \"$APOLLO_GRAPH_REF\" ] && missing_params+=(\"ApolloGraphRef\")\n[ -z \"$SUBGRAPH_NAME\" ] && missing_params+=(\"SubgraphName\")\n[ -z \"$SUBGRAPH_URL\" ] && missing_params+=(\"SubgraphUrl\")\n[ -z \"$SCHEMA\" ] && missing_params+=(\"Schema\")\n\nif [ -n \"$missing_params\" ]; then\n  >&2 echo \"Missing parameters: ${missing_params[@]}\"\n  exit 1\nfi\n\ncurl -sSL https://rover.apollo.dev/nix/$ROVER_VERSION -o installer\n\nif [ \"$(head -n1 installer)\" != \"#!/bin/bash\" ]; then\n  >&2 echo \"There was a problem fetching $ROVER_VERSION of Rover CLI:\"\n  >&2 cat installer\n  rm installer\n  exit 1\nfi\n\nsh installer --force 2>&1\nrm installer\n\nAPOLLO_KEY=$APOLLO_KEY ~/.rover/bin/rover subgraph check $APOLLO_GRAPH_REF \\\n  --name $SUBGRAPH_NAME \\\n  --routing-url $SUBGRAPH_URL \\\n  --schema $SCHEMA \\\n  2>&1\n"
+  },
+  "Parameters": [
+    {
+      "Name": "ApolloRoverVersion",
+      "Label": "Apollo Rover Version",
+      "HelpText": "The version of the Apollo Rover CLI to use for running `rover subgraph publish`.\nMay be either semver (i.e. `0.19.1`), `latest`, or empty",
+      "DefaultValue": "",
+      "DisplaySettings": {
+        "Octopus.ControlType": "SingleLineText"
+      }
+    },
+    {
+      "Name": "ApolloKey",
+      "Label": "Apollo Key",
+      "HelpText": "The API key to Apollo Studio",
+      "DefaultValue": "#{APOLLO_KEY}",
```

Comment:
As above - it is best to leave the default value blank rather than reference a possibly non-existent variable.